packages/node_modules/@webex/plugin-meetings/src/locus-info/parser.js

```diff
@@ -1,118 +1,706 @@
-import {min} from 'lodash';
+import {difference} from 'lodash';
 
-import LocusUtil from '../locus-info/utils';
-import {DELTAEVENT, LOCUS} from '../constants';
+import SimpleQueue from '../common/queue';
+import LoggerProxy from '../common/logs/logger-proxy';
 
-const Parser = {};
+/**
+ * Locus Delta Parser
+ * @private
+ * https://sqbu-github.cisco.com/WebExSquared/cloud-apps/wiki/Locus-Delta-Events
+ */
+export default class Parser {
+  // processing status
+  static status = {
+    IDLE: 'IDLE',
+    PAUSED: 'PAUSED',
+    WORKING: 'WORKING'
+  }
 
-// https://sqbu-github.cisco.com/WebExSquared/cloud-apps/wiki/Locus-Delta-Events
-Parser.updateWithDeltaLocus = (deltaLocus, meeting) => {
-  if (!(deltaLocus.sequence && (deltaLocus.sequence.rangeStart <= deltaLocus.sequence.rangeEnd) && (deltaLocus.sequence.rangeEnd <= min(deltaLocus.sequence.entries)))) {
-    return null;
+  // loci comparison states
+  static loci = {
```

Comment:
I like this use of constants mapped to the class definition.

There is a safe way to make these immutable as well:

**parser.constants.js**
```js
const CONSTANTS = {
  STATUS: {
    IDLE: 'IDLE',
    /* ... */
  },
  LOCI: {
    EQ: 'EQUAL',
    /* ... */
  }
};

export default CONSTANTS;
```

**parser.js**
```js
import CONSTANTS from './parser.constants.js';
/* ... */
export default class Parser {
  /* ... */
  static get status() { // getter to prevent user assigning values.
    return { ...CONSTANTS.STATUS }; // spread operator to generate a new, immutable object.
  }
  /* ... */
}
```

This will prevent a user from changing the values of `Parser.status` values at runtime and isolate constants to a specific file for this class. You could also just place the constants declaration above the class, but can be problematic for some documentation extractors/generators.

We should discuss this type of constant declaration and accessor for future implementations in other areas.

Reply:
This class is an internal class that's private, there are no users that would be consuming it. Personally I wanted to keep the constants / "enums" contained in the class.

Reply:
Great suggestion though thanks, I think in future for classes that aren't private using `getter` makes a lot of sense.